keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/DeliveryArtifact.kt

```diff
@@ -35,11 +35,15 @@ interface DeliveryArtifact {
   val name: String
```

Comment:
Artifacts now have two additional fields: `reference` and `deliveryConfigName`. Reference is a friendly name that you define to refer to your artifact later. It defaults to the "name".

We auto-populate delivery config name when you submit your delivery config. I thought about making a submitted and an actual type for the artifacts, but doing it this way allows us to more easily roll this out (existing artifacts don't have a delivery config name, so we auto populate a default, which we will have to then migrate).

Reply:
Do we also exclude `deliveryConfigName` from exported JSON/YAML?

Reply:
What happens when an artifact is referenced by multiple delivery configs?

Reply:
@robfletcher there's not great/any support for exporting an artifact, yet. If you mean within the json when you ask for the delivery config, I definitely don't remove it. That would probably be a good improvement.

@asher it'll be a separate "artifact" in our database - that way, two configs can care about, say, the same package but different release statuses. The "delivery artifact" becomes more a way to sort and filter versions. The actual full list of versions is stored with just artifact name and type.

Reply:
That is what I meant, yes

Reply:
@robfletcher made serialization better so that delivery config name doesn't show

Reply:
is nice